### PR TITLE
Add docker host config support

### PR DESCRIPTION
**Pull request by anish**
Add ability to specify host configuration while creating a
docker latent worker. Currently regularly used options like --net and --dns cannot be provided while creating a docker latent worker, resulting in a broken worker as this often breaks network access

A new hostconfig parameter is added that accepts host config options as a dict, complete list of supported options can be found [here](https://docker-py.readthedocs.org/en/latest/hostconfig/).

### CODE REVIEW

**Comment by anish** [comment]
@rutsky looks like the CircleCI issue is back

**Comment by rutsky** [comment]
@anish, I restarted job without cache. Hopefully at some point all caches will be recreated and all VMs will be same, and this issue will gone.

**Comment by rutsky** [comment]
@anish, now it fails with actual error in your changes.
Use 
```
`xxx <xxx>`__
``` 
(double underscores at the end) for anonymous links in RST (and please fix other link with which you conflict).

**Comment by anish** [comment]
@rutsky fixed

File: master/buildbot/test/unit/test_worker_docker.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def test_constructor_minimal(self):
 
     def test_constructor_all_docker_parameters(self):
         # Volumes have their own tests
-        bs = self.ConcreteWorker('bot', 'pass', 'unix:///var/run/docker.sock', 'worker_img', ['/bin/sh'], dockerfile="FROM ubuntu", version='1.9', tls=True)
+        bs = self.ConcreteWorker('bot', 'pass', 'unix:///var/run/docker.sock', 'worker_img', ['/bin/sh'], dockerfile="FROM ubuntu", version='1.9', tls=True, hostconfig={'network_mode': 'fake', 'dns': ['1.1.1.1', '1.2.3.4.']})
```
Comment (rutsky) [inline master/buildbot/test/unit/test_worker_docker.py]: Please arrange arguments over several lines, I think in general maximum 80 or 100 characters per line convention is used.

Do you intentionally used invalid IP address string here: `1.2.3.4.` (with dot at the end)?

File: master/docs/manual/cfg-workers-docker.rst [hunk 2]
```diff
@@ -216,6 +216,10 @@ In addition to the arguments available for any :ref:`Latent-Workers`, :class:`Do
     Address of the master the worker should connect to. Use if you master machine does not have proper fqdn.
     This value is passed to the docker image via environment variable ``BUILDMASTER``
 
+``hostconfig``
+    (optional)
+    Extra host configuration parameters passed as a dictionary used to create HostConfig object. See `docker-py's HostConfig documentation <http://docker-py.readthedocs.org/en/latest/hostconfig/>`_ for all the supported options.
+
```
Comment (rutsky) [inline master/docs/manual/cfg-workers-docker.rst]: Please add short notice about this feature into release notes: https://github.com/buildbot/buildbot/blob/master/master/docs/relnotes/index.rst

Also I would appreciate if you include notice about your previous PR #2029 into release notes.

Comment (anish) [inline master/docs/manual/cfg-workers-docker.rst]: Done !

File: master/buildbot/test/unit/test_worker_docker.py (hunk 1 above)
Comment (anish) [inline master/buildbot/test/unit/test_worker_docker.py]: No, I messed up here. Fixed now

File: master/docs/relnotes/index.rst [hunk 3]
```diff
@@ -22,11 +22,14 @@ Features
 
 * :class:`GitPoller` now has a ``buildPushesWithNoCommits`` option to allow the rebuild of already known commits on new branches.
 * Add GitLab authentication plugin for web UI. See :class:`buildbot.www.oauth2.GitLabAuth`.
+* :class:`DockerLatentWorker` now has a ``hostconfig`` parameter that can be used to setup host configuration when creating a new
```
Comment (rutsky) [inline master/docs/relnotes/index.rst]: In RST "one line = one sentence" convention is used in Buildbot (not everywhere yet), please place whole sentence on a single line.

**Comment by anish** [comment]
@rutsky updated

**Comment by rutsky** [comment]
LGTM, thanks!

@delanne, is this PR looks good to you and covers your use case that you implemented in #2038?

**Comment by delanne** [comment]
:+1: lgtm

**Pull request merged**